ksql-rest-app/src/test/resources/ksql-plan-schema/schema-final.json

```diff
@@ -0,0 +1,885 @@
+        },
+        "right" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        }
+      },
+      "title" : "streamTableJoinV1",
+      "required" : [ "@type", "properties", "joinType", "formats", "left", "right" ]
+    },
+    "StreamToTable" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "streamToTableV1" ],
+          "default" : "streamToTableV1"
+        },
+        "source" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "formats" : {
+          "$ref" : "#/definitions/Formats"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        }
+      },
+      "title" : "streamToTableV1",
+      "required" : [ "@type", "source", "formats", "properties" ]
+    },
+    "StreamWindowedAggregate" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "streamWindowedAggregateV1" ],
+          "default" : "streamWindowedAggregateV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "source" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "formats" : {
+          "$ref" : "#/definitions/Formats"
+        },
+        "nonFuncColumnCount" : {
+          "type" : "integer"
+        },
+        "aggregations" : {
+          "type" : "array",
+          "items" : {
+            "type" : "string"
+          }
+        },
+        "windowExpression" : {
+          "type" : "string"
+        }
+      },
+      "title" : "streamWindowedAggregateV1",
+      "required" : [ "@type", "properties", "source", "formats", "nonFuncColumnCount", "aggregations", "windowExpression" ]
+    },
+    "TableAggregate" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "tableAggregateV1" ],
+          "default" : "tableAggregateV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "source" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "formats" : {
+          "$ref" : "#/definitions/Formats"
+        },
+        "nonFuncColumnCount" : {
+          "type" : "integer"
+        },
+        "aggregations" : {
+          "type" : "array",
+          "items" : {
+            "type" : "string"
+          }
+        }
+      },
+      "title" : "tableAggregateV1",
+      "required" : [ "@type", "properties", "source", "formats", "nonFuncColumnCount", "aggregations" ]
+    },
+    "TableFilter" : {
```

Comment:
Q: do we need different steps for `TableFilter` and `SourceFilter`?  I know our _implementation_ has different steps, but I'm not sure the plan does...

Reply:
I think it's useful for the steps to define the type of the result. One benefit (which we admittedly aren't leveraging just yet) is that you could express constraints on what the source of a given step can be. For example, you can only do a windowed aggregate on a stream.